ledger/participant-integration-api/src/main/scala/platform/apiserver/configuration/RateLimitingConfig.scala

```diff
@@ -6,11 +6,24 @@ package com.daml.platform.apiserver.configuration
 final case class RateLimitingConfig(
     maxApiServicesQueueSize: Int,
     maxApiServicesIndexDbQueueSize: Int,
-)
+    maxHeapSpacePercentage: Int,
+    maxOverThresholdZoneSize: Long,
+) {
+  def collectionUsageThreshold(maxPoolBytes: Long): Long = {
+    val percentageBasedThreshold = (maxHeapSpacePercentage * maxPoolBytes) / 100
+    val zoneBasedThreshold = maxPoolBytes - maxOverThresholdZoneSize
+    Math.max(percentageBasedThreshold, zoneBasedThreshold)
+  }
+}
 
 case object RateLimitingConfig {
-  val default: RateLimitingConfig = RateLimitingConfig(
+
+  val Megabyte: Long = 1024L * 1024L
+
+  val Default: RateLimitingConfig = RateLimitingConfig(
     maxApiServicesQueueSize = 10000,
     maxApiServicesIndexDbQueueSize = 1000,
+    maxHeapSpacePercentage = 85,
+    maxOverThresholdZoneSize = 100 * Megabyte,
```

Comment:
In general, I like the fact that this safety buffer is on the small-ish side of the scale. It means that for most practical installations the absolute size will be dominating over the relative one. 
Now I am starting to think that it is maybe too aggressive. We will basically leave 100MB safety buffer whenever the pool is >=667MB.

How about making it 300MB. It would mean that it will start dominating for pools >=2GB